### DISCUSSION TITLE
Transactions from f4 addresses

### DISCUSSION THREAD

**Discussion by generaten** [filecoin-project/community#636] [Q&A]
Hi! I am writing an application, part of the functionality of which is as follows: creating accounts of type f2 and f4, as well as sending transactions. f4 account is created as follows: first, an eth address with public and private keys is created. Then eth is converted to an f4 address. But there was a problem. The message sending code (using the MpoolPush method) works fine when I send a transaction from f2 to f2 or from f2 to f4, but crashes with the 'signature did not match' error when I try to send a transaction from f4 to f2. I have not changed the private key in any way (it is still generated for eth). In this regard, the question is: when converting a 0x address to an f4 address, do you need to do something with the private key? Or is it possible that when sending a message from f4 to f2, the signature is somehow different? I would also be grateful if you could tell me if it is possible to use MpoolPush for transactions from f4 to f4, and if possible, how.

**Comment by longfeiWan9** [comment]
> when converting a 0x address to an f4 address, do you need to do something with the private key?

No, you do not need to do anything about the private key for your 0x address. Just convert 0x address to f4 address.

And it is possible to use mpoolPush to push transactions to Lotus nodes. Any txs can be pushed using mpoolPush. The issue must be related to the transaction construction for sending FIL from f4 to f2. Can you share some detail on how you create that tx ?
